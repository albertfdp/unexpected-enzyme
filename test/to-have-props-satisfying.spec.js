import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class User extends Component {
  render() {
    const { name, email } = this.props;

    return (
      <div className="user">
        {name}
        <span className="email">{email}</span>
      </div>
    );
  }
}

User.propTypes = {
  name: PropTypes.string,
  email: PropTypes.string
};

describe('to-have-props-satisfying', () => {
  it('passes when the passed matches the expected', () => {
    expect(
      mount(<User name="Harriet Banks" email="harriet.banks@kinetar.com" />),
      'to have props satisfying',
      {
        name: /Banks$/,
        email: 'harriet.banks@kinetar.com'
      }
    );
  });

  it('fails when the actual does not match the expected', () => {
    expect(
      () =>
        expect(
          mount(
            <User name="Harriet Banks" email="harriet.banks@kinetar.com" />
          ),
          'to have props satisfying',
          {
            name: /Foo$/,
            email: 'harriet.banks@kinetar.com'
          }
        ),
      'with error matching snapshot'
    );
  });
});
