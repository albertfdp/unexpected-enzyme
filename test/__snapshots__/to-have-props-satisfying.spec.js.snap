// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-have-props-satisfying fails when the actual does not match the expected 1`] = `
"expected <User name=\\"Harriet Banks\\" email=\\"harriet.banks@kinetar.com\\" /> to have props satisfying { name: /Foo$/, email: 'harriet.banks@kinetar.com' }

{
  name: 'Harriet Banks', // should match /Foo$/
  email: 'harriet.banks@kinetar.com'
}"
`;
